File: 0011-container-with-most-water/0011-container-with-most-water.ts
function maxArea(height: number[]): number {
    let l = 0;
    let r = height.length-1; 
    let max = 0 
    while (l < r) {
        let curr = (r - l) * Math.min(height[l], height[r])
        max = Math.max(max, curr)

        if (height[l] < height[r]) {
            l++;
        } else {
            r--;
        }
    }
    return max
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        left = 0
        right = len(height)-1
        ans = 0
        while left < right:
            vol = (right-left) * min(height[left], height[right])
            ans = max(ans, vol)

            if height[left] > height[right]:
                right -= 1
            else:
                left += 1

        return ans
            